doc/sphinx/language/core/records.rst

```diff
@@ -177,6 +177,37 @@ Defining record types
          Class MyClass := { myfield2 : nat }.
          About myfield2. (* Argument name defaults to the class name and is marked implicit *)
 
+.. _record_where_clause:
+
+   .. example:: Using a :g:`where` clause in a record field
+
+      Records themselves do not support a :g:`where` clause, but record fields do,
+      with a somewhat unintuitive behaviour: the notation defined in the where clause
+      is active after the field where it appears, not in the field itself.
+
+      .. rocqtop:: all
+
+         Reserved Notation "a & b" (at level 40).
+         Record foo (op : nat -> nat -> nat) := mkfoo
+         {
+           point: nat where "a & b" := (op a b);
+           cond: forall a b, a & b = b & a
+         }.
```

Comment:
How is a record field notation different from non-record field notations?  Does it behave differently?  Why is it useful?  Can you include info that somewhere?  It would helpful to add a line that uses the record field notation.

Also, it would be better if you can come up with a realistic example.  This seems contrived.  I don't understand how `a & b` relates to point, which is merely a nat.

Reply:
I can't because the feature is broken. It has to be contrived since the behaviour of "where clause" is (probably by chance) weird in records. If you want I can add `op` as a field in the Record, but the point that it's not effective in the field where the `where` clause is written must be clear.
It will probably be deprecated in the near future, but we have to document the present.

Reply:
see  https://rocq-prover.zulipchat.com/#narrow/channel/423352-Rocq-Notation-system/topic/.E2.9C.94.20Reserved.20Notation.20for.20custom.20scope for an example